edgelet/edgelet-http/src/lib.rs

```diff
@@ -247,3 +250,29 @@ impl HyperExt for Http {
         })
     }
 }
+
+pub trait UrlExt {
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error>;
+}
+
+impl UrlExt for Url {
+    #[cfg(unix)]
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error> {
+        debug_assert_eq!(self.scheme(), UNIX_SCHEME);
+        self.to_file_path()
+            .map_err(|()| ErrorKind::InvalidUri(self.to_string()).into())
+    }
+
+    #[cfg(windows)]
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error> {
+        // We get better handling of Windows file syntax if we parse a
+        // unix:// URL as a file:// URL.
```

Comment:
Can you add an example to this comment of where the `unix` scheme leads to a bad parse?

Reply:
Yep, will do.